pipeline-maven/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JacocoReportPublisher.java

```diff
@@ -125,182 +115,7 @@ public void process(@NonNull StepContext context, @NonNull Element mavenSpyLogsE
-                    sourceDirectoryRelativePath,
-                    classesDirectoryRelativePath,
-                    excludes,
-                    mavenArtifact.toString() + " " + pluginInvocation));
-        }
-
-        JacocoPublisher jacocoPublisher = new JacocoPublisher();
-
-        String aggregatedDestFile =
-                jacocoReportDetails.stream().map(details -> details.execFile).collect(Collectors.joining(","));
-        String aggregatedSourceDirectory = jacocoReportDetails.stream()
-                .map(details -> details.sourceDirectory)
-                .collect(Collectors.joining(","));
-        String aggregatedClassesDirectory = jacocoReportDetails.stream()
-                .map(details -> details.classesDirectory)
-                .collect(Collectors.joining(","));
-        String aggregatedExclusionPattern = jacocoReportDetails.stream()
-                .map(details -> details.excludes)
-                .filter(e -> e != null && !e.isEmpty())
-                .distinct()
-                .collect(Collectors.joining(","));
-
-        jacocoPublisher.setExecPattern(aggregatedDestFile);
-        jacocoPublisher.setSourcePattern(aggregatedSourceDirectory);
-        jacocoPublisher.setClassPattern(aggregatedClassesDirectory);
-        jacocoPublisher.setExclusionPattern(aggregatedExclusionPattern);
-
-        try {
-            jacocoPublisher.perform(run, workspace, launcher, listener);
-        } catch (Exception e) {
-            listener.error("[withMaven] jacocoPublisher - exception archiving JaCoCo results for " + jacocoReportDetails
-                    + ": " + e);
-            LOGGER.log(Level.WARNING, "Exception processing JaCoCo results", e);
-            throw new MavenPipelinePublisherException(
-                    "jacocoPublisher", "archiving JaCoCo results for " + jacocoReportDetails, e);
-        }
-    }
-
-    public static class JacocoReportDetails {
-        final String execFile, sourceDirectory, classesDirectory, excludes, description;
-
-        public JacocoReportDetails(
-                String execFile, String sourceDirectory, String classesDirectory, String excludes, String description) {
-            this.execFile = execFile;
-            this.sourceDirectory = sourceDirectory;
-            this.classesDirectory = classesDirectory;
-            this.excludes = excludes;
-            this.description = description;
-        }
-
-        @Override
-        public String toString() {
-            return "JacocoReportDetails{" + "execFile='"
-                    + execFile + '\'' + ", sourceDirectory='"
-                    + sourceDirectory + '\'' + ", classesDirectory='"
-                    + classesDirectory + '\'' + ", excludes='"
-                    + excludes + '\'' + ", description='"
-                    + description + '\'' + '}';
-        }
+        listener.getLogger().println("[withMaven] JacocoReportPublisher is deprecated and will be removed in a future version");
```

Comment:
This is a functional loss. You should instead just delete the class here, then move it to the `jacoco` plugin, adding an `optional` dep on `pipeline-maven`. (To avoid cyclic deps, this needs to happen as a two-step process, but you can at least prepare and pretest the changes using incremental versions.)

Reply:
Alternately, to deprecate without replacement, it would be more polite to fail (`AbortException`) with a message explaining what to do instead: stop relying on the magic integration, and run the JaCoCo publisher step (or the corresponding Coverage API plugin step + config, whatever that may be) explicitly, with a suggested fileset tailored to default mojo output patterns (e.g. `**/target/jacoco/whatever-it-is`).